### DISCUSSION TITLE
Improperly formatted armor header because Version header has no value

### DISCUSSION THREAD

**Discussion by camoredo** [openpgpjs/openpgpjs#1556] [Q&A] [answered]
Hi, would just like to ask if there is any way I can bypass the header check for empty value? I'm getting `Error: Improperly formatted armor header: Version:\n` because the header's value is undefined

For reference, here's the PGP message format:

```
-----BEGIN PGP MESSAGE-----
Version:

........
-----END PGP MESSAGE-----
```

Thank you

**Comment by twiss** [comment, marked as answer]
Hi :wave: For now there isn't. But fwiw, the `Version:` header is entirely inconsequential and unused, so as a workaround you can also just remove that line before parsing. We could also make the parsing more lenient, a PR would be welcome.

**Reply by camoredo** [reply]
Thanks for answering, @twiss! I'm planning to create a PR for it. Just to clarify the changes for it, should we allow no value for all headers or only if the header is `Version`?

**Reply by twiss** [reply]
For all headers is fine. Tbh, I would just change this line: https://github.com/openpgpjs/openpgpjs/blob/93644b7c58bbcd94ed364500d26ad0d694d4f937/src/encoding/armor.js#L193 into a debug error like the one below. Note that the guidance on this point changed between RFC4880 and [the crypto refresh](https://datatracker.ietf.org/doc/html/draft-ietf-openpgp-crypto-refresh-06#page-86), so the comment above it can be updated as well to quote the latter instead. (Thanks in advance!)

**Reply by camoredo** [reply]
Got it, I've added a PR here https://github.com/openpgpjs/openpgpjs/pull/1557. Thanks!
